packages/ckeditor5-image/src/imageupload/imageuploadui.js

```diff
@@ -8,56 +8,162 @@
  */
 
 import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import ImageUploadPanelView from './ui/imageuploadpanelview';
+
 import FileDialogButtonView from '@ckeditor/ckeditor5-upload/src/ui/filedialogbuttonview';
+import { createImageTypeRegExp, prepareIntegrations } from './utils';
+
 import imageIcon from '@ckeditor/ckeditor5-core/theme/icons/image.svg';
-import { createImageTypeRegExp } from './utils';
+
+import { isImage } from '../image/utils';
 
 /**
  * The image upload button plugin.
  *
  * For a detailed overview, check the {@glink features/image-upload/image-upload Image upload feature} documentation.
  *
- * Adds the `'imageUpload'` button to the {@link module:ui/componentfactory~ComponentFactory UI component factory}.
+ * Adds the `'imageUpload'` dropdown to the {@link module:ui/componentfactory~ComponentFactory UI component factory}.
  *
  * @extends module:core/plugin~Plugin
  */
 export default class ImageUploadUI extends Plugin {
 	/**
 	 * @inheritDoc
 	 */
-	init() {
+	static get pluginName() {
+		return 'ImageUploadUI';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	afterInit() {
 		const editor = this.editor;
-		const t = editor.t;
+		const command = editor.commands.get( 'imageUpload' );
 
-		// Setup `imageUpload` button.
 		editor.ui.componentFactory.add( 'imageUpload', locale => {
-			const view = new FileDialogButtonView( locale );
-			const command = editor.commands.get( 'imageUpload' );
-			const imageTypes = editor.config.get( 'image.upload.types' );
-			const imageTypesRegExp = createImageTypeRegExp( imageTypes );
+			let imageUploadView;
+			const integrations = prepareIntegrations( editor );
 
-			view.set( {
-				acceptedType: imageTypes.map( type => `image/${ type }` ).join( ',' ),
-				allowMultipleFiles: true
-			} );
+			if ( integrations ) {
+				imageUploadView = new ImageUploadPanelView( locale, { integrations } );
+			} else {
+				imageUploadView = new ImageUploadPanelView( locale );
+			}
```

Comment:
> Are we planning to add additional options to this class?

No idea, but we have the `options` object pattern to do things like that.

Reply:
btw. it's funny because your suggestion is so obvious here. I know why I didn't do this in the first place — because I forgot to check if `integrations` exists, so I always got "cannot iterate over `null`" (or something like that). Slow clap for me 👏 .

Reply:
Also, I had to change the return value of the `prepareIntegrations()` helper to `undefined` because I forgot that `null` doesn't trigger the default param value in ES6.